### PR TITLE
build: add itest-race build target to catch data races with itests

### PR DESCRIPTION
**Pull request by bhandras**
This PR adds `itest-race` build target that can be used with
individual itests to detect and catch uncovered data races.

### CODE REVIEW

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

File: Makefile [hunk 1]
```diff
@@ -143,6 +143,14 @@ build-itest:
 	@$(call print, "Building itest binary for ${backend} backend.")
 	CGO_ENABLED=0 $(GOTEST) -v ./lntest/itest -tags="$(DEV_TAGS) $(RPC_TAGS) rpctest $(backend)" -c -o lntest/itest/itest.test$(EXEC_SUFFIX)
 
+build-itest-race:
+	@$(call print, "Building itest btcd and lnd.")
```
Comment (Crypt-iQ) [inline Makefile:147]: Could print out that we're using data race detector?

File: Makefile [hunk 2]
```diff
@@ -143,6 +143,14 @@ build-itest:
...
+	CGO_ENABLED=0 $(GOBUILD) -tags="rpctest" -o lntest/itest/btcd-itest$(EXEC_SUFFIX) $(ITEST_LDFLAGS) $(BTCD_PKG)
+	CGO_ENABLED=1 $(GOBUILD) -race -tags="$(ITEST_TAGS)" -o lntest/itest/lnd-itest$(EXEC_SUFFIX) $(ITEST_LDFLAGS) $(PKG)/cmd/lnd
```
Comment (Crypt-iQ) [inline Makefile:149]: Q: why is CGO_ENABLED=1?

**Review by Crypt-iQ — COMMENTED** [review]
I think we'd have to bump the timeouts because this introduces a significant slowdown? Is bumping at the command-line possible?

File: Makefile (hunk 2 above)
Comment (bhandras) [reply Makefile:149]: Go requires `CGO_ENABLED=1` for `race` builds.

File: Makefile (hunk 1 above)
Comment (bhandras) [reply Makefile]: done

**Comment by bhandras** [comment]
> I think we'd have to bump the timeouts because this introduces a significant slowdown? Is bumping at the command-line possible?

Yes timeouts may probably be different, but the intention is to use the `itest-race` target for localized bug hunting at first and then later on we'll integrate with the CI once our itests are less flaky overall.

**Review by Crypt-iQ — APPROVED** [review]

**Pull request merged**